### PR TITLE
[gbinder] Handle changing the rpc protocol during execution

### PR DESCRIPTION
**Pull request by aleasto**
When calling gbinder_ipc_new with a different protocol for the same
device node, we would end up getting the old cached GBinderIpc object
referencing the old protocol. Make sure to update it.

I'm not sure if this is an acceptable solution, or if we should instead recreate the object and replace it in the hash table.

ServiceManager already handles this by using different hash tables for the different GBinderServiceManagerClass.

### CODE REVIEW

**Comment by aleasto** [comment]
Was having issues with the previous version. This works better. Still don't know if it's correct.

File: src/gbinder_ipc.c [hunk 1]
```diff
@@ -1869,15 +1880,16 @@ gbinder_ipc_new(
...
             self->priv->object_registry.io = gbinder_driver_io(driver);
             /* gbinder_ipc_dispose will remove iself from the table */
             if (!gbinder_ipc_table) {
                 gbinder_ipc_table = g_hash_table_new(g_str_hash, g_str_equal);
             }
             g_hash_table_replace(gbinder_ipc_table, priv->key, self);
             /* With "/dev/" prefix, it may be too long to be a thread name */
-            priv->name = priv->key +
+            priv->name = self->dev +
                 (g_str_has_prefix(priv->key, "/dev/") ? 5 : 0);
```
Comment (monich) [inline src/gbinder_ipc.c:1893]:
```c
        } else {
            g_free(key);
```

**Comment by monich** [comment]
OK, thanks!

**Comment by monich** [comment]
FYI: c1ff25f (I slightly amended your commit)

**Pull request merged**